src/poetry/repositories/installed_repository.py

```diff
@@ -201,8 +201,10 @@ def create_package_from_pep610(cls, distribution: metadata.Distribution) -> Pack
             # VCS distribution
             source_type = url_reference["vcs_info"]["vcs"]
             source_url = url_reference["url"]
-            source_reference = url_reference["vcs_info"]["requested_revision"]
             source_resolved_reference = url_reference["vcs_info"]["commit_id"]
+            source_reference = url_reference["vcs_info"].get(
+                "requested_revision", source_resolved_reference
+            )
```

Comment:
What's the reason for not setting `source_reference` to `None` if `requested_revision` is not available?

Reply:
The rationale was that we do not want unexpected runtime failures when something references this. But it also seems is the least worst behavior.